public/js/DroidEntities.js

```diff
@@ -232,37 +216,49 @@ class BasicDroid extends AbstractDroid {
      * to target point + max argument
      */
     shootRandom(targetX, targetY, maxWidth, maxHeight) {
-        var randTargetX = targetX + (maxWidth * Math.random());
-        var randTargetY = targetY + ((-maxHeight) * Math.random());
+        var randTargetX = targetX + (maxWidth * Math.random()) - (maxWidth * Math.random());
+        var randTargetY = targetY + (maxHeight * Math.random()) - (maxHeight * Math.random());
         this.shoot(randTargetX, randTargetY);
     }
     /*
      * calculate movement so that it will try to fly around the location of the 
      * target.
      */
-    calcMovement(target) {
-        //if the droid is to the left of target point, then increase the deltaX
-        //by the x velocity
-        if (this.x < target.x) {
+    calcMovement(targetLeft, targetRight) {
+
+        //if the droid is to the left of targetRight and right of targetLeft
+        if (this.x <= targetRight.x && this.x >= targetLeft.x) {
+            //if the droid is moving right, then increase x velocity
+            if (this.deltaX >= 0) {
+                if (this.deltaX < BASIC_DROID_X_MOVEMENT_SPEED)
+                    this.deltaX += BASIC_DROID_X_VELOCITY;
```

Comment:
Can we change `BASIC_DROID_X_VELOCITY` to be something like `BASIC_DROID_X_ACCELERATION` since thats what this variable is representing. And could we also change these to be dependent on the game timer and not the number of ticks, like we've been doing in other movement patterns.
For example:
make the constant be per second, instead of per tick, so like 60 pixels / second instead of 1 pixel / tick.  
Then do `this.deltaX += BASIC_DROID_X_ACCELERATION * this.game.clockTick;`

This eliminates the volatility of the ticks per second.